### ISSUE TITLE
BE Flake: `fetch-logs-test`

### ISSUE THREAD

**Issue by nemanjaglumac** [metabase/metabase#32356]
## Error
```
ERROR in metabase-enterprise.advanced-config.api.logs-test/fetch-logs-test (AFn.java:148)
Uncaught exception, not in assertion.

java.lang.StackOverflowError: 
```

### Postgres
- https://github.com/metabase/metabase/actions/runs/5517779878/jobs/10060878523?pr=32173#step:4:408
- https://github.com/metabase/metabase/actions/runs/5386767028/jobs/9777220183?pr=31887#step:4:410
- https://github.com/metabase/metabase/actions/runs/5478588617/jobs/9979271493#step:4:410

### MySQL
- https://github.com/metabase/metabase/actions/runs/5445139775/jobs/9903968768?pr=32084#step:4:411

**Comment by camsaul** [comment]
Can you post the actual stacktrace?

**Comment by camsaul** [comment]
$5 says this is something like https://metaboat.slack.com/archives/C04DN5VRQM6/p1678140869079749

**Comment by nemanjaglumac** [comment]
@camsaul the links should take you to the exact line with the error and a stack trace in the logs.

LMK if that works.

**Comment by camsaul** [comment]
Ok, I think these is the actual relevant frames

```
metabase.db.setup/build-around-method-default setup.clj:  196
metabase.models.permissions-group-membership/each-row-fn-primary-method-toucan-query-type-insert-*-model-PermissionsGroupMembership/fn permissions_group_membership.clj:   77
metabase.db.connection/do-with-transaction-around-method-java-sql-Connection connection.clj:  145
metabase.db.connection/do-with-transaction-around-method-java-sql-Connection connection.clj:  145
metabase.db.connection/do-with-transaction-around-method-java-sql-Connection connection.clj:  145
metabase.models.user/each-row-fn-primary-method-toucan-query-type-insert-*-model-User/fn user.clj:  102
```

This seems like almost the same exact problem @qnkhuat fixed in #30814, but in this case it's happening when inserting a User rather than updating one. Altho the PermsGroupMembership code does not recursively trigger the use update method, so I'm a little confused as to why this is causing a Stack Overflow TBH

**Comment by camsaul** [comment]
I think this might fix the issue https://github.com/metabase/metabase/pull/32375
